### DISCUSSION TITLE
How to implement a case-insensitive exclusive constraint (for example, for usernames)?

### DISCUSSION THREAD

**Discussion by elliotwaite** [geldata/gel#6448] [Q&A] [answered]
I want to store usernames in a user table but enforce that the usernames are unique in a case-insensitive way. When using Postgres directly, I used the [citext](https://www.postgresql.org/docs/current/citext.html) extension to accomplish this (I set the username column's type to CITEXT and added a UNIQUE constraint to it). Until the citext extension is supported in EdgeDB, do you have any suggestions for the best way of accomplishing this?

I could create a separate `username_lower` column that stores all usernames in lowercase format, and then add an exclusive constraint to that column, but I was wondering if there was a better solution.

**Comment by notramo** [comment, marked as answer]
https://www.edgedb.com/docs/stdlib/constraints#constraint::std::exclusive

Is it possible to use something like `constraint exclusive on (str_lower(.username))`?

**Reply by elliotwaite** [reply]
@notramo, I never got around to trying this. But thanks for the suggestion.
